src/main/java/org/opensearch/ad/rest/handler/AbstractAnomalyDetectorActionHandler.java

```diff
@@ -231,11 +231,36 @@ public void start() {
             createOrUpdateDetector();
             return;
         }
+        // if only validate API called then check if any errors occur for custom result index
+        // and wrap them as ADValidationException so they are properly returned as ADValidationResponse
+        // also make sure not to create actual index when validating if it doesn't exist
+        if (this.isDryRun) {
```

Comment:
Only check permission when result index exists ? `If (dryRun && anomalyDetectionIndices.doesIndexExist(resultIndex))` ?

Reply:
At that line I would know that resultIndex exists cause the if check above it checks if its null and if its null it proceeds right away to `createOrUpdateDetector`